Plugin/LSP/LSPManager.cpp

```diff
@@ -1206,35 +1218,79 @@ void Manager::OnDocumentSymbolsForHighlight(LSPEvent& event)
     }
     CHECK_PTR_RET(editor);
 
-    const auto& symbols = event.GetSymbolsInformation();
     wxString classes, variables, methods, others;
-    for (const auto& si : symbols) {
-        switch (si.GetKind()) {
-        case LSP::kSK_Module:
-        case LSP::kSK_Namespace:
-        case LSP::kSK_Package:
-        case LSP::kSK_Class:
-        case LSP::kSK_Enum:
-        case LSP::kSK_Interface:
-        case LSP::kSK_Struct:
-        case LSP::kSK_Object:
-            classes << si.GetName() << " ";
-            break;
-        case LSP::kSK_Method:
-        case LSP::kSK_Function:
-            methods << si.GetName() << " ";
-            break;
-        case LSP::kSK_TypeParameter:
-            others << si.GetName() << " ";
-            break;
-        case LSP::kSK_EnumMember:
-        case LSP::kSK_Property:
-        case LSP::kSK_Field:
-        case LSP::kSK_Variable:
-        case LSP::kSK_Constant:
-            variables << si.GetName() << " ";
-        default:
-            break;
+    
+    // lambda to recursively parse a DocumentSymbol hierarchy
+    std::function<void(const DocumentSymbol&)> parseDocumentSymbolRec = [&](const DocumentSymbol& symbol) {        
```

Comment:
Project is in C++20, but for information in C++23, there is [deducing this/explicit object parameter](https://en.cppreference.com/w/cpp/language/function.html#Explicit_object_parameter) which allow recursive lambda (without relying in `std::function` member):
```
auto parseRec = [](this const auto& self, const DocumentSymbol& symbol){ /*..*/ self(child); };
```